16_microbiome_community.Rmd

```diff
@@ -229,22 +229,26 @@ if(!require(bluster)){
 
 The algorithm used is "short random walks"[@Pons2006]. 
 Graph is constructed using different k values (the number of nearest neighbors to consider during graph construction) 
-using the clr assay data. Then plotting the communities using the earlier PCoA ordination as a visual exploration aid.
+using the clr assay data. Then plotting the communities using UMAP ordination as a visual exploration aid.
 
 ```{r, message=FALSE, warning=FALSE}
 library(bluster)
 library(patchwork) # For arranging several plots as a grid
+library(scater)
+
+# Calculating the log transformation required by UMAP ordination 
+tse <- transformCounts(tse, method = "log10", pseudocount = 1)
+# Performing and storing UMAP
+tse <- runUMAP(tse, name="UMAP", exprs_values="log10")
 
 k <- c(2,3,5,10)
 ClustAndPlot <- function(x) {
   # Creating the graph and running the short random walks algorithm  
   graph_clusters <- clusterRows(t(assays(tse)$clr), NNGraphParam(k=x))
-  euclidean_pcoa_df$communities <- graph_clusters
   
-  # Results of the clustering as a color for each sample  
-  ggplot(data = euclidean_pcoa_df, aes(x=pcoa1, y=pcoa2, color = communities)) +
-    geom_point() +
-    labs(x = "PC1", y = "PC2", title = paste0("k = ", x))
+  # Results of the clustering as a color for each sample
+  plotUMAP(tse, colour_by = I(graph_clusters)) +
+    labs(title = paste0("k = ", x))
```

Comment:
I find it somehow confusing that the ordination (UMAP) and clustering (NNGraph) are done in different domains (log10 vs. clr). Does UMAP strictly require log10p transformation, or could it be operate with clr as well?  It would be better for consistency to have both ordination & clustering operating on the same data.

Reply:
At first look at documentation I thought it was strictly logcounts. However it is not, can be done with clr as well. I change it.

Reply:
Now as method "rclr" has been implemented perhaps you could use that?